pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/tests/Test_TNCuda_Var.cpp

```diff
@@ -23,139 +23,173 @@
 
 #include <catch2/catch.hpp>
 
+#include "DevTag.hpp"
+#include "ExaTNCuda.hpp"
 #include "MPSTNCuda.hpp"
 #include "MeasurementsTNCuda.hpp"
 #include "TNCudaGateCache.hpp"
 #include "cuda_helpers.hpp"
 
+#include "TestHelpersTNCuda.hpp"
+
 /// @cond DEV
 namespace {
 using namespace Pennylane::LightningTensor::TNCuda::Measures;
 using namespace Pennylane::LightningTensor::TNCuda::Observables;
 } // namespace
 /// @endcond
 
-TEMPLATE_TEST_CASE("Test variance of NamedObs", "[MPSTNCuda_Var]", float,
-                   double) {
-    using TensorNetT = MPSTNCuda<TestType>;
-    using NamedObsT = NamedObsTNCuda<TensorNetT>;
+TEMPLATE_LIST_TEST_CASE("Test variance of NamedObs", "[TNCuda_Var]",
+                        TestTNBackends) {
+    using TNDeviceT = TestType;
+    using PrecisionT = typename TNDeviceT::PrecisionT;
+    using NamedObsT = NamedObsTNCuda<TNDeviceT>;
 
     std::size_t bondDim = GENERATE(2, 3, 4, 5);
     std::size_t num_qubits = 2;
     std::size_t maxBondDim = bondDim;
+    DevTag<int> dev_tag{0, 0};
```

Comment:
Could we move `DevTag` to the `createTNState` method?

Reply:
Done!